internal/shellgen/tmpl/glibc-patch.nix.tmpl

```diff
@@ -0,0 +1,64 @@
+{
+  description = "Patches packages to use a newer version of glibc";
+
+  inputs = {
+    nixpkgs-glibc.url = "{{ .NixpkgsGlibcFlakeRef }}";
+
+    {{- range $name, $flakeref := .Inputs }}
+    {{ $name }}.url = "{{ $flakeref }}";
+    {{- end }}
+  };
+
+  outputs = args@{ self, nixpkgs-glibc {{- range $name, $_ := .Inputs -}}, {{ $name }} {{- end }} }:
+    {{ with .Outputs -}}
+    let
+      # Initialize each nixpkgs input into a new attribute set with the
+      # schema "pkgs.<input>.<system>.<package>".
+      #
+      # Example: pkgs.nixpkgs-80c24e.x86_64-linux.python37
+      pkgs = builtins.mapAttrs (name: flake:
+        if builtins.hasAttr "legacyPackages" flake then
+          {
+            {{- range $system, $_ := .Packages }}
+            {{ $system }} = (import flake {
+              system = "{{ $system }}";
+              config.allowUnfree = true;
+              config.allowInsecurePredicate = pkg: true;
```

Comment:
would be nice to have this respect the `boolean` value from the `devpkg.Package.AllowInsecure()`